controllers/productsController.js

```diff
@@ -72,10 +76,98 @@ module.exports.doCreate = (request, response) => {
+    if (request.session.userId) {
+        // redirect to dashboard
+        response.redirect('/dashboard');
+    }
+    try {
+        const product = await Product.findById(productId).exec();
+
+        if (product) {
+            response.render('product-update', {
+                site_title: SITE_TITLE,
+                title: 'Product Update',
+                product: product
+            });
+        } else {
+            //redirect to 404 page
+            return response
+                .status(err.status || 404)
+                .render('404', { err: err });
+        }
+    } catch (err) {
+        //redirect to 500 page
+        return response
+            .status(err.status || 500)
+            .render('500', { err: err });
+    }
+}
+module.exports.doUpdate = (request, response) => {
+    var upload = multer({
+        storage: fileUpload.files.storage(),
+        allowedFile: fileUpload.files.allowedFile
+    }).single('image');
+    upload(request, response, async function (err) {
+        // Checking if the error is an instance of MulterError, which would indicate 
+        // an error specifically related to the file upload process, e.g. 
+        // the file is too large, no file was attached, etc.
+        if (err instanceof multer.MulterError) {
+            // Sending the multer error to the client
+            return response
+                .status(err.status || 500)
+                .render('500', { err: err });
+        } else if (err) { // If there's another kind of error (not a MulterError), then handle it here
+            // Sending the generic error to the client
+            return response
+                .status(err.status || 500)
+                .render('500', { err: err });
+        } else { // If no errors occurred during the file upload, continue to the next step
+            const imageUrl = `/public/uploads/${request.file.filename}`;
+            const productId = request.params.productId
+            try {
+                // this is the new updated product properties
+                const updatedData = {
+                    name: request.body.name,
+                    description: request.body.description,
+                    price: request.body.price,
+                    note: request.body.note,
+                    category: request.body.category,
+                    stockQuantity: request.body.stockQuantity,
+                    imageURL: imageUrl,
+                    isAvailable: request.body.isAvailable == 'on'
+                };
+                // functions both as find (productId) and update (updateData).
+                const updatedProduct = await Product.findByIdAndUpdate(productId, updatedData, {
+                    new: true  // This option returns the updated document
+                })
+                if (updatedProduct) {
+                    // redirect back to the product page
+                    response.redirect("/product");
```

Comment:
It should redirect me back to the product that I just updated.
This should be similar to the create method.